src/test/java/io/confluent/connect/elasticsearch/integration/ElasticSearchSinkConnectorIT.java

```diff
@@ -0,0 +1,315 @@
+
+    // Wait Connector to write data into elastic-search
+    waitForConnectorToWriteDataIntoES(
+        CONNECTOR_NAME,
+        Integer.valueOf(MAX_TASKS),
+        KAFKA_TOPIC,
+        NUM_RECORDS);
+    assertRecordsCountAndContent(NUM_RECORDS, KAFKA_TOPIC);
+
+    int status = changeESpassword("elastic", "elastic1");
+    Assert.assertEquals(200, status);
+
+    sendTestDataToKafka(NUM_RECORDS, NUM_RECORDS);
+    totalRecords = connect.kafka().consume(
+        NUM_RECORDS * 2,
+        CONSUME_MAX_DURATION_MS,
+        KAFKA_TOPIC);
+    log.info("Number of records added in kafka {}", totalRecords.count());
+    // Refresh ES client with new credential at test level to perform search operation
+    refreshESClient();
+    // Second batch of records should not be entertained due to credential change
+    waitForConnectorToWriteDataIntoES(
+        CONNECTOR_NAME,
+        Integer.valueOf(MAX_TASKS),
+        KAFKA_TOPIC,
+        NUM_RECORDS);
+    assertRecordsCountAndContent(NUM_RECORDS, KAFKA_TOPIC);
+    // Task must fail after credential change.
+    waitForConnectorTaskToFail();
+  }
+
+  @Test
+  public void testPermissionChange() throws InterruptedException, IOException {
+    int status = createUserWithPermission(
+        "test_user",
+        "test_password",
+        "superuser");
+    Assert.assertEquals(200, status);
+
+    // Run the connector with dummy user
+    props.put(ElasticsearchSinkConnectorConfig.CONNECTION_USERNAME_CONFIG, "test_user");
+    props.put(ElasticsearchSinkConnectorConfig.CONNECTION_PASSWORD_CONFIG, "test_password");
+    sendTestDataToKafka(0, NUM_RECORDS);
+    ConsumerRecords<byte[], byte[]> totalRecords = connect.kafka().consume(
+        NUM_RECORDS,
+        CONSUME_MAX_DURATION_MS,
+        KAFKA_TOPIC);
+    log.info("Number of records added in kafka {}", totalRecords.count());
+
+    // Configure Connector and wait some specific time to start the connector.
+    connect.configureConnector(CONNECTOR_NAME, props);
+    waitForConnectorToStart(CONNECTOR_NAME, Integer.valueOf(MAX_TASKS));
+
+    // Wait Connector to write data into elastic-search
+    waitForConnectorToWriteDataIntoES(
+        CONNECTOR_NAME,
+        Integer.valueOf(MAX_TASKS),
+        KAFKA_TOPIC,
+        NUM_RECORDS);
+    assertRecordsCountAndContent(NUM_RECORDS, KAFKA_TOPIC);
+
+    // Update the 'dummy_user' with read permissions
+    status = createUserWithPermission(
+        "test_user",
+        "test_password",
+        "monitoring_user");
+    Assert.assertEquals(200, status);
+
+    sendTestDataToKafka(NUM_RECORDS, NUM_RECORDS);
+    totalRecords = connect.kafka().consume(
+        NUM_RECORDS * 2,
+        CONSUME_MAX_DURATION_MS,
+        KAFKA_TOPIC);
+    log.info("Number of records added in kafka {}", totalRecords.count());
+
+    // Second batch of records should not be entertained due to change in permissions
+    waitForConnectorToWriteDataIntoES(
```

Comment:
This is for asserting the initially inserted records. It won't guarantee you that second batch is not inserted. Either you can remove this part or alter it to check 2nd batch of records which are inserted ko kafka after changing ES password is not present in Elastic Search index.